### PR TITLE
Update kanister build v0.0.20

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

update kanister build to ~~v0.0.19~~ v0.0.20. This updates to go 1.19. It also updates the linter and as leads to several changes across all packages.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Review by julio-lopez — APPROVED** [review]

**Review by viveksinghggits — APPROVED** [review]

File: pkg/blockstorage/vmware/vmware.go [hunk 1]
```diff
@@ -791,6 +791,7 @@ func (ge govmomiError) Format() string {
 	return fmt.Sprintf("[%s]", strings.Join(msgs, "; "))
 }
 
+// nolint:gocognit
```
Comment (miquella) [inline pkg/blockstorage/vmware/vmware.go:794]: "Directive" type comments shouldn't have a space after the `//`.

```suggestion
//nolint:gocognit
```

File: pkg/envdir/envdir.go [hunk 2]
```diff
@@ -28,10 +29,18 @@ func EnvDir(dir string) ([]string, error) {
...
 		return nil, errors.Wrap(err, "failed to read env from dir:"+dir)
 	}
+	fis := make([]fs.FileInfo, 0, len(entries))
+	for _, entry := range entries {
+		fi, err := entry.Info()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to read info from entry:"+entry.Name())
+		}
+		fis = append(fis, fi)
+	}
```
Comment (miquella) [inline pkg/envdir/envdir.go:43]: Can you explain why we're making this change? I don't quite understand.

File: pkg/kube/pod.go [hunk 3]
```diff
@@ -235,9 +234,9 @@ func checkNodesStatus(p *v1.Pod, cli kubernetes.Interface) error {
 }
 
 // checkPVCAndPVStatus does the following:
-// 1. if PVC is present then check the status of PVC
-// 	1.1 if PVC is pending then check if the PV status is VolumeFailed return error if so. if not then wait for timeout.
-// 2. if PVC not present then wait for timeout
+//  1. if PVC is present then check the status of PVC
+//     1.1 if PVC is pending then check if the PV status is VolumeFailed return error if so. if not then wait for timeout.
+//  2. if PVC not present then wait for timeout
```
Comment (miquella) [inline pkg/kube/pod.go:239]: This reformatting is showing that Go doesn't recognize the nested "1.1" item (Go doesn't support nested lists at all). The way this is formatted, it's actually treating it as a continuation of the `1.` list item — "… the status of PVC 1.1 if PVC is …".

We should probably reformat this to not use a nested list (i.e. move "1.1" to be a top-level item in the list, renumbering the items or switching to an unordered list).

File: pkg/kube/pod.go [hunk 4]
```diff
@@ -248,9 +247,9 @@ func getVolStatus(ctx context.Context, p *v1.Pod, cli kubernetes.Interface, name
...
```
Comment (miquella) [inline pkg/kube/pod.go:252]: Ditto, as above.

File: pkg/blockstorage/ibm/ibmcloud.go [hunk 5]
```diff
@@ -124,7 +124,7 @@ func (s *ibmCloud) volumeParse(ctx context.Context, vol *ibmprov.Volume) (*block
 	}
 	attribs[LunIDAttName] = vol.LunID
 
-	if len(vol.IscsiTargetIPAddresses) < 0 {
+	if len(vol.IscsiTargetIPAddresses) == 0 {
```
[the remark this answers is not in the document]
Comment (bathina2) [reply pkg/blockstorage/ibm/ibmcloud.go:127]: https://github.com/kanisterio/kanister

File: pkg/blockstorage/vmware/vmware.go (hunk 1 above)
Comment (bathina2) [reply pkg/blockstorage/vmware/vmware.go:794]: heh, theres a bunch of places where this needs to be fixed

File: pkg/envdir/envdir.go (hunk 2 above)
Comment (bathina2) [reply pkg/envdir/envdir.go:43]: This deprecation note explains it - https://pkg.go.dev/io/ioutil#ReadDir

Comment (bathina2) [reply pkg/envdir/envdir.go:43]: However I blindly converted. After looking at the code more, I can probably use the response from ReadDir directly.

**Comment by julio-lopez** [comment]
@bathina2 other than Mark's comments, LGTM.

**Review by julio-lopez — APPROVED** [review]
@miquella had additional comments. Please follow up in separate PRs.

File: pkg/kube/pod.go [hunk 6]
```diff
@@ -235,9 +234,9 @@ func checkNodesStatus(p *v1.Pod, cli kubernetes.Interface) error {
 }
 
 // checkPVCAndPVStatus does the following:
-// 1. if PVC is present then check the status of PVC
-// 	1.1 if PVC is pending then check if the PV status is VolumeFailed return error if so. if not then wait for timeout.
-// 2. if PVC not present then wait for timeout
+// - if PVC is present then check the status of PVC
+// - if PVC is pending then check if the PV status is VolumeFailed return error if so. if not then wait for timeout.
+// - if PVC not present then wait for timeout
```
Comment (miquella) [inline pkg/kube/pod.go:239]: tiny nit: [Go Doc Comments](https://go.dev/doc/comment#lists) says that bullet lists should be indented, so this should be:

```suggestion
//   - if PVC is present then check the status of PVC
//   - if PVC is pending then check if the PV status is VolumeFailed return error if so. if not then wait for timeout.
//   - if PVC not present then wait for timeout
```

File: pkg/kube/pod.go [hunk 7]
```diff
@@ -248,9 +247,9 @@ func getVolStatus(ctx context.Context, p *v1.Pod, cli kubernetes.Interface, name
...
```
Comment (miquella) [inline pkg/kube/pod.go:252]: ditto:

```suggestion
//   - if PVC is present then check the status of PVC
//   - if PVC is pending then check if the PV status is VolumeFailed return error if so. if not then wait for timeout.
//   - if PVC not present then wait for timeout
```

File: pkg/restic/restic.go [hunk 8]
```diff
@@ -471,7 +471,8 @@ func SnapshotIDsFromSnapshotCommand(output string) ([]string, error) {
 // SpaceFreedFromPruneLog gets the space freed from the prune log output
 // For reference, here is the logging command from restic codebase:
 // Verbosef("will delete %d packs and rewrite %d packs, this frees %s\n",
-//		len(removePacks), len(rewritePacks), formatBytes(uint64(removeBytes)))
+//
+//	len(removePacks), len(rewritePacks), formatBytes(uint64(removeBytes)))
```
Comment (miquella) [inline pkg/restic/restic.go:475]: tiny nit: this looks like it's supposed to be formatted:

```suggestion
//
//	Verbosef("will delete %d packs and rewrite %d packs, this frees %s\n",
//		len(removePacks), len(rewritePacks), formatBytes(uint64(removeBytes)))
```

**Review by miquella — APPROVED** [review]
LGTM, just a few tiny Go Doc Comment nits

**Pull request merged**